prime-router/docs/api/lookup-tables-openapi.yaml

```diff
@@ -0,0 +1,162 @@
+
+security:
+  - OktaAuth: [ ]
+
+paths:
+  /api/lookuptables:
+    get:
+      description: Fetch the list lookup tables
+      parameters:
+        - name: showAll
+          in: query
+          description: Set to true to include inactive tables
+          schema:
+            type: boolean
+      responses:
+        '200':
+          description: The list of tables with version information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableVersionInfo'
+        '400':
+          description: if there was an error fetching the list of tables
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+
+  /api/lookuptables/{tableName}:
+    get:
+      description: Fetch the latest version of a table
+      responses:
+        '200':
+          description: The table data
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DynamicTableRows'
+        '404':
+          description: The table name was not found
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+        '400':
+          description: if there was an error fetching the table data
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+    post:
+      description: Create a new version of a lookup table
+      requestBody:
+        description: Optional description in *Markdown*
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DynamicTableRows'
+      responses:
+        '200':
+          description: The table was created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableVersionInfo'
+        '400':
+          description: if there was an error creating the table
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+
+  /api/lookuptables/{tableName}/{tableVersion}:
+    get:
+      description: Fetch a specific version of a table
+      responses:
+        '200':
+          description: The table data
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DynamicTableRows'
+        '404':
+          description: The table name and/or version was not found
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+        '400':
+          description: if there was an error fetching the table data
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+    post:
+      description: Activate a specific version of a table
```

Comment:
How does deactivation of a table version occur?
Is there a rule that only one version of a table can be active at a time?
If so, does activating a version also deactivate all other versions?

Reply:
yes, only one version at a time is allowed, and it is enforced via a database constraint.  In the code on the other PR you can see how I deactivate and activate within the same transaction.

Reply:
What is the content for this post?

Reply:
This is really a REST-style comment. Reading REST posts should have a representation of the state change that you want to effect. In other words, you should have a content part that represents the `isActive` state.